app/webpacker/controllers/select_customer_controller.js

```diff
@@ -0,0 +1,113 @@
+import TomSelectController from "./tom_select_controller";
+
+export default class extends TomSelectController {
```

Comment:
I'm wondering if there is a better way than inheritance here. A common pattern in OOP is to prefer composition over inheritance. Would it make sense to have this controller search for customers and then just pass the data to tom-select which could be separate?

And I noticed that some other parts of the code use the DependantSelectController to solve the state-country problem. Would it be better to use that instead of `setValueOnTomSelectController`?

I guess that this is the kind of thing that Matt meant by using several re-usable methods instead of creating something new for each problem. But I still haven't worked with Stimulus and can only guess.

Reply:
I didn't know about `DependantSelectController`, I probably should have a look. 

Regarding composition vs. inheritance, you mean that this part of code: 
```js
fetch("/admin/search/customers.json?" + new URLSearchParams(params))
      .then((response) => response.json())
      .then((json) => {
        this.items = json;
        callback(json);
      })
      .catch((error) => {
        this.items = [];
        console.log(error);
        callback();
      });
```
which search for a user and then call the callback() should be in an other place in order:
 - to be re-used by other tom-select controller
 - to re-use the controller tom-select that display customer (with `renderOption`) 

Right?

I'll make that change too.

Reply:
I think that `DependantSelectController` is probably not so usefull in our case. It used the `data-dependant-select-options-value` attribute, and is more or less too much static than needed.

Reply:
Okay, I'm relying on your assessment here.

Reply:
I've created two mixins:
- `useRenderCustomer` that is in charge of rendering a customer
- `useSearchCustomer` that is in charge of search for a customer
Both of those mixins can be reused in other controllers that want to display for a customer or search for a customer.

See commits 8fec5f1464b43ff72275db3880bbee9b880c8345, 39b7f56e6d245293ea66883d0353b5a25e78318b. 

What do you think?